tests/dogtag/pytest-ansible/provision/osp_provision.py

```diff
@@ -0,0 +1,206 @@
+import os
+import random
+import sys
+
+import click
+import openstack
+
+
+OS_AUTH_URL = os.getenv('OS_AUTH_URL')
+OS_PROJECT_NAME = os.getenv('OS_PROJECT_NAME')
+OS_USERNAME = os.getenv('OS_USERNAME')
+OS_REGION_NAME = os.getenv('OS_REGION_NAME')
+OS_KEYPAIR_NAME = os.getenv('OS_KEYPAIR_NAME')
+CI_FLAVOR_NAME = os.getenv('CI_FLAVOR_NAME')
+DOMAIN_NAME = os.getenv('DOMAIN_NAME')
+OS_PASSWORD = os.getenv('OS_PASSWORD')
+OS_NETWORK_ID = os.getenv('OS_NETWORK_ID').split(',')
+# OS_NETWORK_ID = random.choice(OS_NETWORK_ID)
+
+inventory_file = "ansible_inventory"
+instance_record = '/tmp/instance_record.txt'
+
+
+def create_connection(auth_url, region, project_name, username, password):
+    return openstack.connect(auth_url=auth_url,
+                             project_name=project_name,
+                             username=username,
+                             password=password,
+                             user_domain_name=DOMAIN_NAME,
+                             project_domain_name=DOMAIN_NAME,
+                             region_name=region)
+
+
+def spawn_server(conn, image_name, server_type='master'):
+    """
+    This method spawns a server for provided image.
+    The method looks for available network and also sets an inventory which is used in teardown state.
+    """
+
+    print("Spawning server: ")
+    set_available_network(conn)
+
+    if server_type == 'master':
+        server_name = 'OSP_API_Master_{}'.format(random.randint(999, 99999))
+    else:
+        server_name = 'OSP_API_Client_{}'.format(random.randint(999, 99999))
+    image = conn.compute.find_image(image_name)
+    flavor = conn.compute.find_flavor(CI_FLAVOR_NAME)
+    try:
+        server = conn.compute.create_server(name=server_name, image_id=image.id,
+                                            flavor_id=flavor.id, networks=[{"uuid": OS_NETWORK_ID}],
+                                            key_name=OS_KEYPAIR_NAME)
+
+
+        # Writing server details in the inventory file which is used at the time of teardown.
+        if os.path.isfile(instance_record):
+            with open(instance_record, 'a') as f:
+                f.write('{}\n'.format(server.name))
+        else:
+            with open(instance_record, 'w') as f:
+                f.write('{}\n'.format(server.name))
+
+        # Wait time for VM to come active is 10 min
+        server = conn.compute.wait_for_server(server, status='ACTIVE', failures='ERROR', interval=2, wait=600)
+        return server
+    except Exception as e:
```

Comment:
You shouldn't need a try/except here. Since you're catching the exception and exiting, it'd be better if you just let the exception go (and propagate up). Then you get pretty stack traces... :)

Reply:
fixed